geomstats/geometry/poincare_ball.py

```diff
@@ -317,3 +317,268 @@ def inner_product_matrix(self, base_point=None):
         identity = gs.eye(self.dimension, self.dimension)
 
         return gs.einsum('i,jk->ijk', lambda_base, identity)
+
+
+NORMALIZATION_FACTOR_CST = gs.sqrt(gs.pi / 2)
+PI_2_3 = pow((2 * gs.pi), 2 / 3)
+SQRT_2 = gs.sqrt(2.)
+
+
+class GaussianDistribution():
+    """A class for Gaussian distributions."""
+
+    @staticmethod
+    def gaussian_pdf(data, means, variances, norm_func, metric):
+        """Return the probability density function."""
+        data_length, _, n_gaussian = data.shape + (means.shape[0],)
+
+        data_expanded = gs.expand_dims(data, 1)
+        data_expanded = gs.repeat(data_expanded, n_gaussian, axis=1)
+
+        means_expanded = gs.expand_dims(means, 0)
+        means_expanded = gs.repeat(means_expanded, data_length, axis=0)
+
+        variances_expanded = gs.expand_dims(variances, 0)
+        variances_expanded = gs.repeat(variances_expanded, data_length, 0)
+
+        num = gs.exp(-(metric.dist(data_expanded, means_expanded) ** 2)
+                     / (2 * variances_expanded ** 2))
+
+        den = norm_func(variances)
+
+        den = gs.expand_dims(den, 0)
+        den = gs.repeat(den, data_length, axis=0)
+
+        result = num / den
+
+        return result
```

Comment:
Is this of shape = (n_data,)? Following new vectorization logic?

Reply:
It is of shape (n_data, n_gaussian)

Reply:
Ok great! Could you also add the Parameters/Returns description to make this clearer? Thanks!